src/sentry/static/sentry/app/views/settings/components/forms/form.tsx

```diff
@@ -137,6 +138,12 @@ export default class Form extends React.Component<Props> {
     };
   }
 
+  componentDidUpdate(prevProps: Props) {
+    if (!isEqual(prevProps.initialData, this.props.initialData)) {
+      this.model.setInitialData(this.props.initialData);
+    }
+  }
```

Comment:
We can avoid this if we add a `key` to the <Form> in `organizationSecurityAndPrivacy`